### DISCUSSION TITLE
Code simplification using library that allows Optional.map

### DISCUSSION THREAD

**Discussion by lavafroth** [archlinux/archinstall#1898] [Ideas]
I recently came across [this](https://pypi.org/project/option/) library which enables Rust like semantics for optional and result types.

This might be very useful for the project. Take the example of the optional object `something`, it can either have some object value or `None`. We have a lot of the code that looks like the following:

```py
if something:
    something_else = some_action(something)
else:
    something_else = None
```

which could be simplified to the following using this library:

```py
something_else = something.map(some_action)
```

**Comment by Torxed** [comment]
It's for sure an interesting concept.
Really my only demand on this code base is that it should be accessible to all form of contributors.
So as long as the concept and or code is broadly understandable and not to esoteric *(aside from the sometimes complicated nature of installing the OS itself)*.

I've been doing this for too long that I consider the logic you present pretty understandable, even your `something = something or {}` *"magic"* is pretty nifty to avoid if/else.

But I wonder if `if: x` is more readable/understandable for a broader audience?

tl;dr: I like it, but simplicity is more important than efficiency if that makes sense.

**Reply by lavafroth** [reply]
Absolutely. Also the fact that I come from a Rust background might have had play on this, lol. I think you stance on the readability is very appropriate here. Thanks!

**Comment by svartkanin** [comment]
I don't think introducing a library that imports new coding paradigms is a good idea. Python as a language already comes with the well known Zen "There should be one-- and preferably only one --obvious way to do it.". 

Maybe it'd worth considering something like this if it was a start or early stage of a project, but given that the codebase isn't small and already quite a while around it would be quite a drastic change and add another layer of complexity to it. 

Don't get me wrong I really like the concept of returned error codes over exceptions and it does make tge code sacer and predictable
